### ISSUE TITLE
Remove support to fetch values of a resolved configuration with oconf binary

### ISSUE THREAD

**Issue by gustavnikolaj** [One-com/node-oconf#18]
As per #14 `oconf file a.b.c` will be deprecated.

# v2:
- [x] Add deprecation warning when using `oconf file a.b.c` and release a new minor version

# v3:
- [x] Remove support for `oconf file a.b.c`
- [x] See if any of the current functionality in `oconf file a.b.c` needs to put into `oconf-extract-value`

**Comment by gustavnikolaj** [comment]
```json
{
  "foo": "bar"
}
```

oconf will normally return the JSON.stringified value of an extracted option.

```
$ oconf file.cjson foo
"bar"
```

Where oconf-extract-option will return the raw value.
```
$ oconf-extract-option file.cjson foo
bar
```

oconf would do the same when passed the --bare option
```
$ oconf --bare file.cjson foo
bar
```

oconf-extract-option will keep it's default for compatabilitys sake, and I'll introduce an option `--json` to get the JSON compatible output.

**Comment by gustavnikolaj** [comment]
If you ask for an non existent key, both tools will error out. But oconf had a switch called --allowmissing that would cause it to exit with a status of 0 and no output when a missing key was encountered. This proved useful when using it to configure builds, so I'll add this option to oconf-extract-value too.

**Comment by gustavnikolaj** [comment]
Turns out that oconf-extract-option already had the --json switch. Sorry about that - it's really sunday huh? :-)

**Comment by gustavnikolaj** [comment]
Landed as 7cc2881b3d33887da1258295db0a8ce15be6c87c and will be released with 3.0.0
